### ISSUE TITLE
TensorFlow.keras写基本的神经网络 分类图片（云型）

### ISSUE THREAD

**Issue by ml-xia** [ml-xia/Blog#1]

**Comment by ml-xia** [comment]
https://tensorflow.google.cn/tutorials/keras/basic_classification 
最基本的图像分类教程步骤



# TensorFlow and tf.keras
import tensorflow as tf
from tensorflow import keras
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import OneHotEncoder
# Helper libraries
import numpy as np
import matplotlib.pyplot as plt
import os
from skimage import io
import pandas as pd
import random

l0=64;w0=64

def filelist(dir):   #返回指定目录下(包含文件夹)所有文件（不包含文件路径）
    f=[]
    for root,subdirs,files in os.walk(dir):
        for name in files:
            f.append(dir+name)
    return f  

def load(path):
    f=filelist(path);X=[];fn=[]
    for fname in f:
        try:
            img=io.imread(fname)
            img=img.reshape((img.size,))
            if len(img)!=l0*w0*3:
                print(fname);print(img.shape);continue
            X.append(img);fn.append(fname.split('\\')[-1])
        except:
            print(fname)
    return np.array(X),fn

def mean_f1(y_true, y_pred):
    f1list=[]
    for k in range(5):
        print(y_true);print(y_pred);print(k);input()
        recall=tf.metrics.recall_at_k(y_true,y_pred,k)
        precision=tf.metrics.precision_at_k(y_true,y_pred,k)
        f1=recall*precision*2/(recall+precision)
        print(f1);input()
        f1list.append(f1)
    f1avg=sum(f1list)/len(f1list)
    return tf.Variable(f1avg)

def m_f1(y_true, y_pred):
    print(y_true);print(y_pred);input()
    recall=tf.metrics.recall(y_true,y_pred)
    precision=tf.metrics.precision(y_true,y_pred)
    print(recall);print(precision);input()
    f1=tf.Variable(recall*precision*2/(recall+precision))
    print(f1);input()
    return f1


rootdir='D:\\sq\\'
class_names = ['Gaoji', 'Juan', 'Chenji', 'Ji', 'Jia']
X_train,fn_train=load(rootdir+'train\\yasuo\\')
X_train = MinMaxScaler().fit_transform(X_train)
y_train=pd.read_csv(rootdir+'train\\train.csv',index_col=0)
y_train=y_train.loc[fn_train].values-1
X_train = X_train.reshape(-1,w0,l0,3)
plt.figure(figsize=(10,10))
for i in range(25):
    plt.subplot(5,5,i+1)
    plt.xticks([])
    plt.yticks([])
    plt.grid(False)
    plt.imshow(X_train[i], cmap=plt.cm.binary)
    plt.xlabel(class_names[int(y_train[i])])
plt.show()
#X_train = X_train/255
print(X_train);print(X_train.shape);print(y_train.shape);print('import X Y ok!');input()


########划分数据集########
#dic_train=dict(zip(fn_train,X_train))
#random.shuffle(fn_train);y_train=y_train.loc[fn_train].values-1
#dic_train=np.array([dic_train[fn] for fn in fn_train])
#print(dic_train.shape);print('random ok!');input()
#X_train0=dic_train[:round(len(dic_train)*1)]
#y_train0=y_train[:round(len(y_train)*1),:]
#X_train1=dic_train[round(len(dic_train)*1):]
#y_train1=y_train[round(len(y_train)*1):,:]
#print('set train x y ok!');input()


model = keras.Sequential([
    keras.layers.Flatten(input_shape=(64, 64,3)),
    keras.layers.Dense(64, activation=tf.nn.relu),
    keras.layers.Dense(5, activation=tf.nn.softmax)
])

model.compile(optimizer=tf.train.AdamOptimizer(), 
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.fit(X_train, y_train, epochs=500)
print('training ok!');input()

test_loss, test_acc = model.evaluate(X_train, y_train)
print('Test accuracy:', test_acc);input()

X_test,fn_test=load(rootdir+'test\\yasuo\\')
#print(max(X_train));print(min(X_train));input()
X_test = MinMaxScaler().fit_transform(X_test)
X_test = X_test.reshape(-1,w0,l0,3)
predictions = model.predict(X_test)
rt=np.argmax(predictions,1)+1
rt=rt.reshape(-1,1)
print(rt.shape)
submit=pd.read_csv(rootdir+'submit_example.csv',index_col=0)
#submit.loc[:]=None
submit.loc[fn_test]=rt
submit.to_csv(rootdir+'submit.csv')
print('submit ok!')
